lib/puppet/util/windows/user.rb

```diff
@@ -52,9 +52,20 @@ def check_token_membership
   module_function :check_token_membership
 
   def password_is?(name, password)
-    logon_user(name, password) { |token| }
-  rescue Puppet::Util::Windows::Error
-    false
+    # Windows API LogonUserW always disallows this with error 1327:
+    # Logon failure: user account restriction. Possible reasons are blank passwords
+    # not allowed, logon hour restrictions, or a policy restriction has been enforced.
+    if password.nil? || password == ''
+      raise Puppet::Util::Windows::Error.new(
+        "Windows does not have the ability to verify NULL or empty passwords"
+      )
+    end
```

Comment:
Correct, and thanks for asking.

One weird thing that I did notice though is that the warning is not showing:

```
bundle exec puppet apply -e "user { 'ramon': managehome => false, password => '' }"
Warning: the running version of Bundler is older than the version that created the lockfile. We suggest you upgrade to the latest version of Bundler by running `gem install bundler`.

DL is deprecated, please use Fiddle
Notice: Compiled catalog for vagrant-2008r2.corp.puppetlabs.net in environment production in 0.06 seconds
Notice: /Stage[main]/Main/User[ramon]/ensure: created
Notice: Applied catalog in 0.22 seconds


C:\source\puppet [ticket/stable/PUP-6843-puppet-resource-user-calls-login-with-null-password-on-Windows +63 ~0 -0 !]> net user

User accounts for \\VAGRANT-2008R2

-------------------------------------------------------------------------------
Administrator            bla                      bob
cyg_server               foo_bar                  Guest
ramon                    sshd                     vagrant
The command completed successfully.
```